--- C:/Sandbox/Code/Chrome/extensions/prayer-times-extension/js/pray_time.js	Sat May 21 01:50:08 2011
+++ C:/Sandbox/Code/Chrome/extensions/prayer-times-extension/js/PrayTimes.js	Sat May 21 01:45:42 2011
@@ -5,10 +5,6 @@
 Copyright (C) 2007-2011 PrayTimes.org
 
 Developer: Hamid Zarrabi-Zadeh
-
-Modifications by Mohamed Mansour (2010-2011)
-Source: http://mohamedmansour.com
-
 License: GNU LGPL v3.0
 
 TERMS OF USE:
@@ -72,15 +68,15 @@
 
 	// Time Names
 	timeNames = {
-		imsak    : chrome.i18n.getMessage('imsak'),
-		fajr     : chrome.i18n.getMessage('fajr'),
-		sunrise  : chrome.i18n.getMessage('sunrise'),
-		dhuhr    : chrome.i18n.getMessage('dhuhr'),
-		asr      : chrome.i18n.getMessage('asr'),
-		sunset   : chrome.i18n.getMessage('sunset'),
-		maghrib  : chrome.i18n.getMessage('maghrib'),
-		isha     : chrome.i18n.getMessage('isha'),
-		midnight : chrome.i18n.getMessage('midnight')
+		imsak    : 'Imsak',
+		fajr     : 'Fajr',
+		sunrise  : 'Sunrise',
+		dhuhr    : 'Dhuhr',
+		asr      : 'Asr',
+		sunset   : 'Sunset',
+		maghrib  : 'Maghrib',
+		isha     : 'Isha',
+		midnight : 'Midnight'
 	},
 
 
@@ -166,11 +162,8 @@
 	},
 
 	timeFormat = '24h',
-	timeSuffixes = [
-		chrome.i18n.getMessage('am'),
-		chrome.i18n.getMessage('pm')
-	],
-	InvalidTime =  '-----',
+	timeSuffixes = ['am', 'pm'],
+	invalidTime =  '-----',
 
 	numIterations = 1,
 	offset = {},
@@ -252,8 +245,7 @@
 		lng = 1* coords[1];
 		elv = coords[2] ? 1* coords[2] : 0;
 		timeFormat = format || timeFormat;
-		// Chrome Fix for bug crbug.com/78256
-		// if (date.constructor === Date)
+		if (date.constructor === Date)
 		date = [date.getFullYear(), date.getMonth()+ 1, date.getDate()];
 		if (typeof(timezone) == 'undefined' || timezone == 'auto')
 			timezone = this.getTimeZone(date);
@@ -543,28 +535,8 @@
 	// add a leading 0 if necessary
 	twoDigitsFormat: function(num) {
 		return (num <10) ? '0'+ num : num;
-	},
-
-	//---------------------- Mohamed Extension Update -----------------------
-	
-	// returns the time names.
-	getTimeNames: function() {
-		return timeNames;
-	},
-
-	// returns the method name.
-	getMethodName: function(method) {
-		if (methods[method]) {
-			return methods[method].name;
-		} else {
-			return InvalidTime;
 		}
-	},
   
-	// set time format method.
-	setTimeFormat: function(format) {
-		timeFormat = format;
-	}
 }}
 
 
@@ -601,4 +573,5 @@
 //---------------------- Init Object -----------------------
 
 
-var prayTimes = new PrayTimes('Jafari');
+var prayTimes = new PrayTimes();
+
